lab_2

ArtemShev opened the pull request.

ArtemShev committed:
lab_2

ArtemShev committed:
lab_2

qerq90 commented on prog_lang/lab_2/classes/file_service.py:
```diff
@@ -0,0 +1,67 @@
+    def del_file(self,id):
+        try:
+            Filename = self.data.pop(id)
+            return print(f"{Filename} delete success")
```
зачем возвращать результат print-a

qerq90 commented on prog_lang/lab_2/classes/file_service.py:
```diff
@@ -0,0 +1,67 @@
+        except:
+            raise FileNotFoundError
+
+    def change_dict(self,id, new_id):
```
неудачное название для функции

qerq90 commented on prog_lang/lab_2/classes/file_service.py:
```diff
@@ -0,0 +1,67 @@
+
+    def change_dict(self,id, new_id):
+        try:
+            self.data[new_id] = self.data.pop(id)
```
Что если в `self.data[new_id`] лежало что-то уже?

qerq90 commented on prog_lang/lab_2/classes/file_service.py:
```diff
@@ -0,0 +1,67 @@
+    def get_few_files(self, array_of_id):
+        new_arr = []
+        for one_id in array_of_id:
+            new_arr.append(self.data[one_id])
```
Хотя бы один неправильный id-шник и оно сломается

qerq90 commented on prog_lang/lab_2/classes/file_service.py:
```diff
@@ -0,0 +1,67 @@
+        handler = open('backup.txt', 'w')
+        handler.write(data_str)
+        handler.close()
+    def reincarnate(self):
```
После reincarnate я могу получить сервис с данными, которые буду перезатирать, потому что max_id все еще 0
Плюс, зачем эта функция, если в init можно передать данные просто?

qerq90 commented on prog_lang/lab_2/classes/file_service.py:
```diff
@@ -0,0 +1,67 @@
+
+
+
+while True:
```
Ты либо при импорте получишь бесконечный цикл, блочащий работу всей программы, либо он просто не будет работать.

qerq90 reviewed (changes requested):
Переделать чуток

ArtemShev committed:
final

Исправил названия
Теперь есть проверка в выдаче нескольких файлов
recover(бывш. reincarnate) теперь перезаписывает max_id и айди не перезатираются
